rainier-core/src/main/scala/com/stripe/rainier/ir/ApplyMethodGenerator.scala

```diff
@@ -5,12 +5,13 @@ final private case class ApplyMethodGenerator(classPrefix: String,
                                               outputMethods: Seq[Int])
     extends MethodGenerator {
   val isStatic: Boolean = false
-  val methodName: String = "apply"
-  val methodDesc: String = "([D[D[D)V"
+  val methodName: String = "output"
+  val methodDesc: String = "([D[DI)D"
 
-  outputMethods.zipWithIndex.foreach {
-    case (m, i) =>
-      storeOutput(i) { callExprMethod(m) }
+  loadOutputIndex()
+  tableSwitch(outputMethods) {
+    case Some(m) => callExprMethod(m)
+    case None    => constant(0.0)
```

Comment:
isn't reaching this case a logical error and an exception should be thrown as a fail-fast mechanism?

Reply:
Yes. I'm being lazy here because I haven't looked into how to generate bytecode for an exception yet.